expressions_api/expression_verify.go

```diff
@@ -0,0 +1,130 @@
+	var sqlStatement string
+	switch expressionVerificationInput.expressionDetail.Type {
+	case "CHECK":
+		sqlStatement = fmt.Sprintf("SELECT 1 from %s where %s;", expressionVerificationInput.expressionDetail.ReferenceElement.Name, expressionVerificationInput.expressionDetail.Expression)
+	case "DEFAULT":
+		sqlStatement = fmt.Sprintf("SELECT CAST(%s) as %s", expressionVerificationInput.expressionDetail.Expression, expressionVerificationInput.expressionDetail.ReferenceElement.Name)
+	default:
+		return common.TaskResult[internal.ExpressionVerificationOutput]{Result: internal.ExpressionVerificationOutput{Result: false, Err: fmt.Errorf("invalid expression type requested")}, Err: nil}
+	}
+	result, err := ev.SpannerAccessor.ValidateDML(context.Background(), sqlStatement)
+	return common.TaskResult[internal.ExpressionVerificationOutput]{Result: internal.ExpressionVerificationOutput{Result: result, Err: err, ExpressionDetail: expressionVerificationInput.expressionDetail}, Err: nil}
+}
+
+func (ev *ExpressionVerificationAccessorImpl) validateRequest(verifyExpressionsInput internal.VerifyExpressionsInput) error {
+	if verifyExpressionsInput.Project == "" || verifyExpressionsInput.Instance == "" || verifyExpressionsInput.Conv == nil || verifyExpressionsInput.Source == "" {
+		return fmt.Errorf("one of project, instance, conv or source is empty. These are mandatory fields = %v", verifyExpressionsInput)
+	}
+	for _, expressionDetail := range verifyExpressionsInput.ExpressionDetailList {
+		if expressionDetail.ExpressionId == "" || expressionDetail.Expression == "" || expressionDetail.Type == "" || expressionDetail.ReferenceElement.Name == "" {
+			return fmt.Errorf("one of expressionId, expression, type or referenceElement.Name is empty. These are mandatory fields = %v", expressionDetail)
+		}
+	}
+	return nil
+}
+
+// We simplify conv to remove any existing expressions that are part of the SpSchema to ensure that the stagingDB creation
+// does not fail due to inconsistent, user configured expressions during a schema conversion session.
+// The minimal conv object needed for stagingDB is one which contains all table and column definitions only.
+func (ev *ExpressionVerificationAccessorImpl) simplifyConv(inputConv *internal.Conv) (*internal.Conv, error) {
+	convCopy := &internal.Conv{}
+	convJSON, err := json.Marshal(inputConv)
+	if err != nil {
+		return nil, fmt.Errorf("error marshaling conv: %v", err)
+	}
+	err = json.Unmarshal(convJSON, convCopy)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling conv: %v", err)
+	}
+	// Reset the SpSequences field in the Conv copy.  A nil check
+	// here won't hurt but shouldn't technically be necessary as the
+	// map should always be initialized.
+	if convCopy.SpSequences != nil {
+		convCopy.SpSequences = make(map[string]ddl.Sequence) // Reset to empty map
```

Comment:
Discussed with Astha; we will remove this logic from this PR from now and implement it together with default and check constraint expression exclusion.

Reply:
Done